components/carousel/__tests__/index.test.js

```diff
@@ -78,4 +78,17 @@ describe('Carousel', () => {
     expect(spy).toHaveBeenCalled();
     expect(spy2).toHaveBeenCalledWith('resize', onWindowResized);
   });
+
+  it('warning', () => {
+    const warnSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+    mount(
+      <Carousel vertical>
+        <div />
+      </Carousel>,
+    );
+    expect(warnSpy).toHaveBeenCalledWith(
+      'Warning: [antd: Carousel] `vertical` is deprecated, please use `dotPosition` instead.',
+    );
+    warnSpy.mockRestore();
+  });
```

Comment:
Additional test case for `dotPosition` to check the snapshot since demo snapshot can only check one of them:
```jsx
['left', 'right', 'top', 'bottom'].forEach((dotPosition) => {
  it(dotPosition, () => {
    // ...
  });
});
```